ale/drivers/ody_drivers.py

```diff
@@ -0,0 +1,123 @@
+
+    @property
+    def ephemeris_start_time(self):
+        og_start_time = super().ephemeris_start_time
+        offset = self.label["IsisCube"]["Instrument"]["SpacecraftClockOffset"]
+        if isinstance(offset, pvl._collections.Units):
+            units = offset.units
+            if "ms" in units.lower():
+                offset = offset.value * 0.001
+            else:
+                # if not milliseconds, the units are probably seconds
+                offset = offset.value
+
+        return og_start_time + offset
+
+
+class OdyThemisVisIsisLabelNaifSpiceDriver(LineScanner, IsisLabel, NaifSpice, Driver):
+    """
+
+    """
+
+    @property
+    def instrument_id(self):
+        inst_id = super().instrument_id
+
+        if inst_id not in ["THEMIS_VIS"]:
+            raise Exception(f"{inst_id} is not a valid THEMIS instrument name. Expecting THEMIS_VIS")
+
+        return inst_id
+
+    @property
+    def sensor_model_version(self):
+        return 1
+
+    @property
+    def spacecraft_name(self):
+        name = super().spacecraft_name.replace('_', ' ')
+        if name != "MARS ODYSSEY":
+            raise Exception("{name} for label is not a valid Mars Odyssey spacecraft name")
+        return name
+
+    @property
+    def ikid(self):
+        return self.label['IsisCube']['Kernels']['NaifFrameCode']
+
+    @property
+    def ephemeris_start_time(self):
+        # inst["SpacecraftClockOffset"]
+        # et + offset - ((p_exposureDur / 1000.0) / 2.0);
+        og_start_time = super().ephemeris_start_time
+
+        offset = self.label["IsisCube"]["Instrument"]["SpacecraftClockOffset"]
+        if isinstance(offset, pvl._collections.Units):
+            units = offset.units
+            if "ms" in units.lower():
+                offset = offset.value * 0.001
+            else:
+                # if not milliseconds, the units are probably seconds
+                offset = offset.value
+
+        return og_start_time + offset - (self.line_exposure_duration/2)
+
+    @property
+    def line_exposure_duration(self):
+        """
+        The line exposure duration of the image, in seconds
+
+        Returns
+        -------
+        : float
+          Line exposure duration in seconds
+        """
+        line_exposure_duration = self.label['IsisCube']['Instrument']['ExposureDuration']
+        if isinstance(line_exposure_duration, pvl._collections.Units):
+            units = line_exposure_duration.units
+            if "ms" in units.lower():
+                line_exposure_duration = line_exposure_duration.value * 0.001
+            else:
+                # if not milliseconds, the units are probably seconds
+                line_exposure_duration = line_exposure_duration.value
+        else:
+            # if no units are available, assume the exposure duration is given in milliseconds
+            line_exposure_duration = line_exposure_duration * 0.001
+        return line_exposure_duration
```

Comment:
Can this logic go into the label classes for PDS3 and ISIS?

Reply:
Do you mean the error checking? in which case chasing them all down should be its own PR Or the offset check? I don't know to what extent this is standard in ISIS. If this is standard in ISIS then this implementation should go up to the ISIS label class.

Reply:
Just the check for the millisecond conversion.

Reply:
Yeah, we really should have that in more places but should be its own PR.